homeassistant/components/omnilogic/__init__.py

```diff
@@ -0,0 +1,217 @@
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["sensor"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Omnilogic component."""
+    hass.data.setdefault(DOMAIN, {})
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Omnilogic from a config entry."""
+
+    conf = entry.data
+    username = conf[CONF_USERNAME]
+    password = conf[CONF_PASSWORD]
+    polling_interval = conf[POLL_INTERVAL]
+    api = OmniLogic(username, password)
+
+    try:
+        await api.connect()
+        await api.get_telemetry_data()
+    except LoginException as error:
+        _LOGGER.debug("OmniLogic login error: %s", error)
+        raise PlatformNotReady from error
+    except OmniLogicException as error:
+        _LOGGER.debug("OmniLogic API error: %s", error)
+
+    coordinator = OmniLogicUpdateCoordinator(
+        hass=hass,
+        api=api,
+        name="Omnilogic",
+        polling_interval=polling_interval,
+    )
+    await coordinator.async_refresh()
+
+    if not coordinator.last_update_success:
+        raise ConfigEntryNotReady
+
+    hass.data[DOMAIN][entry.entry_id] = {
+        COORDINATOR: coordinator,
+        OMNI_API: api,
+    }
+
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Unload a config entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, component)
+                for component in PLATFORMS
+            ]
+        )
+    )
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+class OmniLogicUpdateCoordinator(DataUpdateCoordinator):
+    """Class to manage fetching update data from single endpoint."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        api: str,
+        name: str,
+        polling_interval: int,
+    ):
+        """Initialize the global Omnilogic data updater."""
+        self.api = api
+
+        super().__init__(
+            hass=hass,
+            logger=_LOGGER,
+            name=name,
+            update_interval=timedelta(seconds=polling_interval),
+        )
+
+    def update_listeners(self):
+        """Call update on all listeners."""
+        for update_callback in self._listeners:
+            update_callback()
+
+    async def _async_update_data(self):
+        """Fetch data from OmniLogic."""
+        try:
+            _LOGGER.debug("Updating the coordinator data.")
+            data = await self.api.get_telemetry_data()
+            return data
+
+        except OmniLogicException as error:
+            raise UpdateFailed(f"Error updating from OmniLogic: {error}") from error
+
+
+class OmniLogicEntity(CoordinatorEntity):
```

Comment:
Maybe split to a separate file?

Reply:
Done - split out into omnilogic_common.py with the data coordinator for simplicity.